File: watchappwithreactts.client/src/types/movie.ts
export interface MovieSummary {
  id: string;
  title: string;
  thumbnailUrl: string;
  releaseYear: number;
  durationMinutes: number;
  genres: string[];
  rating: number;
}

export interface MovieDetail extends MovieSummary {
  description: string;
  coverUrl: string;
  director: string;
  cast: string[];
  isAvailableInHD: boolean;
}

export interface Pagination {
  currentPage: number;
  totalPages: number;
  totalItems: number;
  limit: number;
}

export interface PaginatedResponse<T> {
  success: boolean;
  data: {
    items: T[];
    pagination: Pagination;
  };
}

export interface MovieDetailResponse {
  success: boolean;
  data: MovieDetail;
}

export interface Subtitle {
  language: string;
  label: string;
  url: string;
}

export interface StreamData {
  movieId: string;
  streamType: string;
  manifestUrl: string;
  subtitles: Subtitle[];
  thumbnailSprites: string;
}

export interface StreamResponse {
  success: boolean;
  data: StreamData;
}

File: WatchAppWithReactTS.Server/Data/Entities/Movie.cs
namespace WatchAppWithReactTS.Server.Data.Entities;

public class Movie
{
    public Guid Id { get; set; }
    public required string Title { get; set; }
    public string? Description { get; set; }
    public required string ThumbnailUrl { get; set; }
    public string? CoverUrl { get; set; }
    public int? ReleaseYear { get; set; }
    public int? DurationMinutes { get; set; }
    public decimal? Rating { get; set; }
    public string? Genres { get; set; }

    public MovieStream? MovieStream { get; set; }
    public ICollection<Subtitle> Subtitles { get; set; } = new List<Subtitle>();
}
